packages/core/src/state/StateSystem.js

```diff
@@ -240,14 +247,25 @@ export default class StateSystem extends System
         this.blendMode = value;
 
         const mode = this.blendModes[value];
+        const gl = this.gl;
 
         if (mode.length === 2)
         {
-            this.gl.blendFunc(mode[0], mode[1]);
+            gl.blendFunc(mode[0], mode[1]);
         }
         else
         {
-            this.gl.blendFuncSeparate(mode[0], mode[1], mode[2], mode[3]);
+            gl.blendFuncSeparate(mode[0], mode[1], mode[2], mode[3]);
+        }
+        if (mode.length === 6)
```

Comment:
can we stick this if inside the else above?

Reply:
no, because there's `else` branch for that if too